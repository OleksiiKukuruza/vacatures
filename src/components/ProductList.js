import React from 'react';
import styled from 'styled-components';
import ProductTableContainer from '../containers/ProductTableContainer';

export const StyledProductList = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  border: 1px solid ${props => props.theme.primary.main};
  border-right: none;
`;

export const StyledHeader = styled.h3`
  color: ${props => props.theme.primary.main};
`;

const ProductList = () => (
  <StyledProductList>
    <StyledHeader>
      Online Vacatures
    </StyledHeader>
    <div>
      Introductiepakketten
    </div>
    <ProductTableContainer />
  </StyledProductList>
);

export default ProductList;
