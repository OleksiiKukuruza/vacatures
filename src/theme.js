const theme = {
  primary: {
    main: '#54b6c8',
    contrastText: '#fff'
  },
  background: '#f1f4f5',
  error: '#f44336',
  spacingUnit: 8
};

export default theme;
