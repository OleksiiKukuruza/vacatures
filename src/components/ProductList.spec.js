import React from 'react';
import { shallow } from 'enzyme';
import theme from '../theme';
import ProductList, { StyledHeader, StyledProductList } from './ProductList';

const setup = (Component) => {
  const wrapper = shallow(<Component theme={theme} />);

  return {
    wrapper
  };
};

describe('ProductList', () => {
  it('renders ProductList without crashing', () => {
    const { wrapper } = setup(ProductList);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders StyledProductList without crashing', () => {
    const { wrapper } = setup(StyledProductList);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders StyledHeader without crashing', () => {
    const { wrapper } = setup(StyledHeader);
    expect(wrapper).toMatchSnapshot();
  });
});
