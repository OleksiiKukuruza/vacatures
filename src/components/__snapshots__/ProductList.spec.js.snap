// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList renders ProductList without crashing 1`] = `
<styled.div>
  <styled.h3>
    Online Vacatures
  </styled.h3>
  <div>
    Introductiepakketten
  </div>
  <Connect(branch(branch(ProductTable))) />
</styled.div>
`;

exports[`ProductList renders StyledHeader without crashing 1`] = `
.c0 {
  color: #54b6c8;
}

<h3
  className="c0"
/>
`;

exports[`ProductList renders StyledProductList without crashing 1`] = `
.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid #54b6c8;
  border-right: none;
}

<div
  className="c0"
/>
`;
